src/_ert/forward_model_runner/job.py

```diff
@@ -193,67 +202,118 @@ def ensure_file_handles_closed():
                 }
-                run_time = dt.now() - run_start_time
-                if (
-                    max_running_minutes is not None
-                    and run_time.seconds > max_running_minutes * 60
-                ):
-                    # If the spawned process is not in the same process group as
-                    # the callee (job_dispatch), we will kill the process group
-                    # explicitly.
-                    #
-                    # Propagating the unsuccessful Exited message will kill the
-                    # callee group. See job_dispatch.py.
-                    process_group_id = os.getpgid(proc.pid)
-                    this_group_id = os.getpgid(os.getpid())
-                    if process_group_id != this_group_id:
-                        os.killpg(process_group_id, signal.SIGKILL)
-
-                    yield Exited(self, exit_code).with_error(
-                        (
-                            f"Job:{self.name()} has been running "
-                            f"for more than {max_running_minutes} "
-                            "minutes - explicitly killed."
-                        )
-                    )
+                if isinstance(exited_msg, Exited):
+                    yield exited_msg
                     return
 
-        exited_message = Exited(self, exit_code)
+        ensure_file_handles_closed([stdin, stdout, stderr])
+        exited_message = self._create_exited_message_based_on_exit_code(
+            max_memory_usage, target_file_mtime, exit_code, fm_step_pids
+        )
+        yield exited_message
+
+    def _create_exited_message_based_on_exit_code(
+        self,
+        max_memory_usage: int,
+        target_file_mtime: Optional[int],
+        exit_code: int,
+        fm_step_pids: Sequence[int],
+    ) -> Exited:
+        # exit_code = proc.returncode
 
         if exit_code != 0:
-            if killed_by_oom(fm_step_pids):
-                yield exited_message.with_error(
-                    f"Forward model step {self.job_data.get('name')} "
-                    "was killed due to out-of-memory. "
-                    "Max memory usage recorded by Ert for the "
-                    f"realization was {max_memory_usage//1024//1024} MB"
-                )
-            else:
-                yield exited_message.with_error(
-                    f"Process exited with status code {exit_code}"
-                )
-            return
+            exited_message = self._create_exited_msg_for_non_zero_exit_code(
+                max_memory_usage, exit_code, fm_step_pids
+            )
+            return exited_message
 
         # exit_code is 0
 
         if self.job_data.get("error_file") and os.path.exists(
             self.job_data["error_file"]
         ):
-            yield exited_message.with_error(
+            exited_message = Exited(self, exit_code)
+            return exited_message.with_error(
```

Comment:
return?? previously we yielded ....

Reply:
Yes, but this is a helper function for creating the message that is being yielded from the calling method.